packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -442,6 +444,11 @@ async fn directory_tree_to_entrypoints_internal(
 ) -> Result<EntrypointsVc> {
     let mut result = IndexMap::new();
 
+    // appDir ignores paths starting with an underscore
+    if directory_name.starts_with('_') {
+        return Ok(Entrypoints(result).cell());
+    }
```

Comment:
We can skip that much earlier in `get_directory_tree` when it's iterating the `entries`. `basename` should not start with `_`